rocketpy/Motor.py

```diff
@@ -0,0 +1,1937 @@
+            thrust starts at time equals 0. If False, no translation
+            is applied.
+
+        Returns
+        -------
+        None
+        """
+        # Retrieve current thrust curve data points
+        timeArray = self.thrust.source[:, 0]
+        thrustArray = self.thrust.source[:, 1]
+        # Move start to time = 0
+        if startAtZero and timeArray[0] != 0:
+            timeArray = timeArray - timeArray[0]
+
+        # Reshape time - set burn time to burnTime
+        self.thrust.source[:, 0] = (burnTime / timeArray[-1]) * timeArray
+        self.burnOutTime = burnTime
+        self.thrust.setInterpolation(self.interpolate)
+
+        # Reshape thrust - set total impulse
+        if oldTotalImpulse is None:
+            oldTotalImpulse = self.evaluateTotalImpulse()
+        self.thrust.source[:, 1] = (totalImpulse / oldTotalImpulse) * thrustArray
+        self.thrust.setInterpolation(self.interpolate)
+
+        # Store total impulse
+        self.totalImpulse = totalImpulse
+
+        # Return reshaped curve
+        return self.thrust
+
+    def evaluateTotalImpulse(self):
+        """Calculates and returns total impulse by numerical
+        integration of the thrust curve in SI units. The value is
+        also stored in self.totalImpulse.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        self.totalImpulse : float
+            Motor total impulse in Ns.
+        """
+        # Calculate total impulse
+        self.totalImpulse = self.thrust.integral(0, self.burnOutTime)
+
+        # Return total impulse
+        return self.totalImpulse
+
+    @abstractproperty
+    def exhaustVelocity(self):
+        """Calculates and returns exhaust velocity by assuming it
+        as a constant. The formula used is total impulse/propellant
+        initial mass. The value is also stored in
+        self.exhaustVelocity.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        self.exhaustVelocity : float
+            Constant gas exhaust velocity of the motor.
+        """
+        pass
+
+    @abstractmethod
+    def evaluateMassDot(self):
+        """Calculates and returns the time derivative of propellant
+        mass by assuming constant exhaust velocity. The formula used
+        is the opposite of thrust divided by exhaust velocity. The
+        result is a function of time, object of the Function class,
+        which is stored in self.massDot.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        self.massDot : Function
+            Time derivative of total propellant mas as a function
+            of time.
+        """
+        pass
+
+    @abstractmethod
+    def evaluateCenterOfMass(self):
+        """Calculates and returns the time derivative of motor center of mass.
+        The result is a function of time, object of the Function class, which is stored in self.yCM.
```

Comment:
Does it compute the time derivative or the relative position? Furthermore, what is this position relative to?

Reply:
This method is only a work-in-progress for now. One of the next few PR's regarding hybrids will tackle it specifically, as well as improve this docstring which I agree it's lacking and confusing. 

The method calculates the position of the center of mass of the total propellant, relative to the **center of mass of the solid propellant**, which is the new reference point for all-things solids and hybrids. 

One of the ways it will do so is by solving an initial value problem of a differential equation (hence the time derivative of the motor's center of mass).